### DISCUSSION TITLE
week 4-  The Prototype

### DISCUSSION THREAD

**Discussion by WeberWr** [The-Escape-Box/TheEscapeBox#27] [General]
### **Welcome back to our project "TheEscapeBox"**

In this post we want to share with you how our prototype looks like and what features the prototype brings with. It basically sums up the work we did in de last weeks and already mentioned in the last few blog posts. But it will be the first blog in which you can see how the game really looks and you can download and play the prototype by yourself.

### Features

_1.  The Player_

At the start we have our player. The player has at this point no animations, but its a first-person so you can't really see the player. The player holds a pistol in the one hand and a flashlight in the other hand. The flashlight is the only light source we have in the game.

For the pistol we have a ammunition system that only allows you to shoot a few times, before you are out of ammo, so use your shots wise.

<img width="450" alt="Player Front" src="https://github.com/The-Escape-Box/TheEscapeBox/assets/97832247/846e6d53-713d-4dff-9a9c-cde224180c52">
<img width="360" alt="Player Side" src="https://github.com/The-Escape-Box/TheEscapeBox/assets/97832247/bb574811-b99e-4d28-b75a-f023b547a85e">

_2. The Demons_

What would be a escape room without demons chasing you. This is why we decided to add for the start the enemies that following you with different speed and different attack damage. So be sure always turn around, they're fast and you don't want to feel their breath on your neck.

As we said in the last blog we use a AI powered mesh navigation agent to follow the player. The mesh can be calculated easy in the unity editor and will leave out walls. 

<img width="420" alt="Demon" src="https://github.com/The-Escape-Box/TheEscapeBox/assets/97832247/0f691d39-1afe-4b5e-b723-1be9308ef61b">
<img width="420" alt="Zombie" src="https://github.com/The-Escape-Box/TheEscapeBox/assets/97832247/475f1234-d47c-407a-8360-33306a3bb8a7">

_3. The Environment_

You also saw the environment before but now you also see what it looks like to stay in it and feel the darkness. To get this feeling we searched through the unity asset store and combined some assets we found there. Overall we are really proud that it looks and feels so scary. 

As you can see the whole environment looks like a giant maze. This is exactly what we want to have the possibility to implement different puzzles. 

<img width="522" alt="Top-Down" src="https://github.com/The-Escape-Box/TheEscapeBox/assets/97832247/86e4c9d5-e171-4f6f-aa3f-ad4a58d54a18">

<img width="1164" alt="Player standing in the environment" src="https://github.com/The-Escape-Box/TheEscapeBox/assets/97832247/9cdd17e8-ac1e-4d8e-a2e9-7453774938a8">

_4. The Puzzle_

The first puzzel is the "Unleash the power of electricity!"
The player need to find the Transformer Box and when he found it, the light in game is going to turn on. But what if it does not go on? Use the Hint!
And when it does and the light is on, do you think you escape the box .... no, n. Now the game start 🚀🚀

<img width="537" alt="The transformer-box" src="https://github.com/The-Escape-Box/TheEscapeBox/assets/97832247/bf8cfffa-5882-4f3e-8ff0-229a659883fa">

<img width="241" alt="The key" src="https://github.com/The-Escape-Box/TheEscapeBox/assets/97832247/fdbe6bb5-f0d3-4228-a103-515f225c06a2">

_5. The Shop_

As last main feature we have the ingame shop. You can open it by pressing "B". Here you can heal yourself, buy ammunition or make your ammunition stronger. You will need this to win the game. The currency is blood (what else should it be in a game like that). You obtain blood by shooting the demons. We plan other ways to obtain blood (any ideas?).

<img width="214" alt="The shop" src="https://github.com/The-Escape-Box/TheEscapeBox/assets/97832247/92607db5-afc9-4389-9cbc-79d00a0ca5a8">

### Play the game here (Press "B" for the shop):

https://mohmad-alnakeshbandi.itch.io/theescapebox

### Time spend
Time we spent this week:
Mohamad: 15h
Lukas: 12h

### Coming in future

- new weapon model with animations
- animations for the player
- refactoring
- more buyable stuff
- enhance the puzzle
- spookier environment
- other ways to obtain blood
- patrolling demons

Feel free to try out the game and tell us about bugs or features you would love to see in the game. We will read all your opinions and try to make decision all of you love.

**Comment by tonoizer** [comment]
Exciting progress! I love the concept of "TheEscapeBox" and can't wait to see how it evolves. Keep up the great work!
